src/core/server/legacy_compat/logging/legacy_logging_server.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { EventEmitter } from 'events';
+// @ts-ignore: implicit any for JS file
+import { Config, transformDeprecations } from '../../../../server/config';
+// @ts-ignore: implicit any for JS file
+import { setupLogging } from '../../../../server/logging';
+import { LogRecord } from '../../logging/log_record';
+
+interface PluginRegisterParams {
+  register: {
+    register: (
+      server: LegacyLoggingServer,
+      options: PluginRegisterParams['options'],
+      cb: () => void
+    ) => void;
+  };
+  options: Record<string, any>;
+}
+
+/**
+ *  The "legacy" Kibana uses Hapi server + even-better plugin to log, so we should
```

Comment:
__note:__ I can theoretically just use `hapi` (v14) instance here, but as I mentioned in comment it feels a bit too much to me (and it's going to be upgraded soon as well), but let me know if you'd rather prefer something else here.